eval/alpaca_farm/run_eval.py

```diff
@@ -0,0 +1,167 @@
+import os
+import json
+import argparse
+import logging
+import random
+import torch
+import datasets
+import vllm
+from alpaca_eval import evaluate as alpaca_farm_evaluate
+from eval.utils import query_openai_chat_model, query_openai_model, generate_completions, dynamic_import_function, load_hf_lm_and_tokenizer
+
+
+def main(args):
+    random.seed(42)
+    os.makedirs(args.save_dir, exist_ok=True)
+
+    logging.info("loading data and model...")
+    alpaca_eval_data = datasets.load_dataset("tatsu-lab/alpaca_farm", "alpaca_farm_evaluation")["eval"]
+    prompts = []
+    chat_formatting_function = dynamic_import_function(args.chat_formatting_function) if args.use_chat_format else None
+    for example in alpaca_eval_data:
```

Comment:
In the original script I had a check for saved generations. Do you think its worth adding this? I found it useful, e.g. in case I ran the code and forgot to have my api key set - I could just rerun the script after setting the key, and skip the (lengthy) generation process.

Reply:
This should be fine, since the generation is pretty quick now with vllm (4 mins for running 70B model on 4 GPUs). Let's keep the code clean.